examples/tests/managed-ng-ami-options/ami-id/index.ts

```diff
@@ -0,0 +1,53 @@
+import * as awsx from '@pulumi/awsx';
+import * as eks from "@pulumi/eks";
+import * as pulumi from "@pulumi/pulumi";
+import * as iam from "../iam";
+import {GetParameterCommand, SSMClient} from "@aws-sdk/client-ssm";
+
+const eksVpc = new awsx.ec2.Vpc("eks-vpc", {
+    enableDnsHostnames: true,
+    cidrBlock: "10.0.0.0/16",
+  });
+
+// IAM roles for the node groups.
+const role = iam.createRole("example-role");
+
+const projectName = pulumi.getProject();
+
+const cluster = new eks.Cluster(`${projectName}`, {
+    skipDefaultNodeGroup: true,
+    deployDashboard: false,
+    vpcId: eksVpc.vpcId,
+    // Public subnets will be used for load balancers
+    publicSubnetIds: eksVpc.publicSubnetIds,
+    // Private subnets will be used for cluster nodes
+    privateSubnetIds: eksVpc.privateSubnetIds,
+    instanceRoles: [role],
+});
+
+// Export the cluster's kubeconfig.
+export const kubeconfig = cluster.kubeconfig;
+
+// Find the recommended AMI for the EKS node group.
+// See https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html
+async function getEksAmiId(k8sVersion: string): Promise<string> {
+    const client = new SSMClient();
```

Comment:
You can do that in pulumi natively. See here for an example:
https://github.com/pulumi/pulumi-eks/blob/9d80b5df0dc491f3dd09c159a0183e40e8350bba/examples/custom-managed-nodegroup/index.ts#L38-L40